### ISSUE TITLE
Provide fasjson-client rpm package

### ISSUE THREAD

**Issue by arrfab** [fedora-infra/fasjson-client#47]
For public consumption, a rpm package would be needed.
Ideally, as we have still people on the following releases/versions interacting with all our infra (Fedora/CentOS combined), we'd build and provide it for :
 
 * Fedora (all current releases)
 * RHEL8/CentOS8
 * RHEL7/CentOS7 (for that one that can be difficult but let's evaluate it at least)

**Comment by abompard** [comment]
I have a beta spec file, if somebody (or future me) wants to start from it : [fasjson-client.spec](https://github.com/fedora-infra/fasjson-client/files/5391131/fasjson-client.spec.txt)

**Comment by nphilipp** [comment]
Notes from convo we had in today's standup:

- not terribly urgent ATM, probably best in a subsequent sprint
- examine feasibility regarding 3rd party Python dependencies in EL7
- `poetry` doesn't seem to be available in EL7/EL8 -> maybe provide a barebones setup.py instead and use setuptools

**Comment by abompard** [comment]
There's a pending review of a dependency to get this packaged: https://bugzilla.redhat.com/show_bug.cgi?id=1878977
Could somebody pick it up? Maybe @nphilipp ?

As for poetry, the tarball uploaded to PyPI contains a `setup.py` that poetry generates, so it should work that way.

**Comment by nphilipp** [comment]
Reviewed here: https://bugzilla.redhat.com/show_bug.cgi?id=1900738
Requested repo here: https://pagure.io/releng/fedora-scm-requests/issue/30874

**Comment by nphilipp** [comment]
Package built for Rawhide: https://koji.fedoraproject.org/koji/buildinfo?buildID=1645295
Branches requested for:
- F33: https://pagure.io/releng/fedora-scm-requests/issue/30875
- F32: https://pagure.io/releng/fedora-scm-requests/issue/30876
- EPEL8: https://pagure.io/releng/fedora-scm-requests/issue/30878

**Comment by nphilipp** [comment]
Built the packages for F32…Rawhide and submitted updates:

- F33: https://bodhi.fedoraproject.org/updates/FEDORA-2020-862267b8d3
- F32: https://bodhi.fedoraproject.org/updates/FEDORA-2020-13c381f3de

In EPEL8 there are still some dependency gaps before the package can be built and used.

- `python-toml` is only available in v. 0.10.0. This could be patched in the RPM for EPEL8 builds.
- `python-bravado` is built but only available in testing. Needs a BRO.
- `python-cryptography` is unavailable. There's the special EPEL-sauce `python3-cryptography` component but neither has an `epel8` branch (the latter has `epel7`).

**Comment by StephenCoady** [comment]
Requested an epel8 branch for `python3-cryptography` and will build and submit updates as soon as its available.

**Comment by StephenCoady** [comment]
The build is failing for `python3-cryptography` without `python3-cffi`.

This package is only on epel7 at the moment so I've requested it to be adopted. I'll get an epel8 build going once I can and then rebuild `python3-cryptography`

**Comment by arrfab** [comment]
python-cryptography *is* built and included in rhe8 : https://git.centos.org/rpms/python-cryptography/tree/c8
So why are you trying to build it through epel8 ? it would be rejected anyway (as it would conflict with the one from BaseOS

**Comment by nphilipp** [comment]
TL;DR: Blocked by `python-rfc3987` not yet being available as an official EPEL8 package.

Something of a more (but not too) detailed status:

* `python-fasjson-client`
  * available in stable repositories (and seemingly usable!) in all active Fedora releases. Version 0.1.1 is still in-flight as testing updates:
    * F32: https://bodhi.fedoraproject.org/updates/FEDORA-2020-6c8881fd70
    * F33: https://bodhi.fedoraproject.org/updates/FEDORA-2020-abaaed01aa
  * available as an [EPEL8 testing update](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2020-207246d65a) along with these dependences:
    * [python-strict-rfc3339](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2020-ab968700c4)
    * [python-swagger-spec-validator](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2020-cf46928158)
    * [python-webcolors](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2020-685648c75e)
  * … `python-rfc3987` is still missing—sent an email to the pkg maintainer requesting co-maintainership last Friday but haven't heard back yet. **⇐ this blocks an official package stack on EL8**
    * built the package (as of current Rawhide) for `epel8-infra` [here](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2020-685648c75e) as a stopgap

**Comment by arrfab** [comment]
Good status report ... I already pointed (for now) some CentOS STG users on my copr repo, which should have same packages (plus `centos-packager` just pulling some other deps) : https://copr.fedorainfracloud.org/coprs/arrfab/fasjson-client/ ..
I'll remove all my builds from that copr once it will be officially in epel8 and f32/f33 .. Also started a discussion to eventually have `centos-packager` itself in Fedora/epel to close the gap
